src/app/core/services/cluster.ts

```diff
@@ -167,21 +172,30 @@ export class ClusterService {
     return this._http.delete(url, {headers: this._headers});
   }
 
-  showDisconnectClusterDialog(cluster: ExternalCluster, projectID: string): Observable<any> {
+  showDisconnectClusterDialog(cluster: ExternalCluster, projectID: string): void {
     const dialogConfig: MatDialogConfig = {
       data: {
         title: 'Disconnect Cluster',
         message: `Are you sure you want to disconnect ${cluster.name} cluster?`,
         confirmLabel: 'Disconnect',
+        throttelButton: true,
+        observable: this._deleteExternalCluster(projectID, cluster.id),
+        // next: this._router.navigate(['/projects/' + projectID + '/clusters'], {
+        //     fragment: `${ClusterListTab.ExternalCluster}`,
+        //   }).finally(() => this._notificationService.success(`Disconnected the ${cluster.name} cluster`))
       },
```

Comment:
if i uncomment lines (183-185) next will execute just when the dialog open before i click the button